custom_components/localtuya/cover.py

```diff
@@ -186,19 +220,194 @@ def __init__(
+            },
+            "name": self._device._friendly_name,
+            "manufacturer": "Tuya Generic",
+            "model": "SmartCover",
+            "sw_version": "3.3",
+        }
+
+    @property
+    def open_cmd(self):
+        """Get name of open command."""
+        _LOGGER.info("def open_cmd called=%s", self._open_cmd)
+        return self._open_cmd
+
+    @property
+    def close_cmd(self):
+        """Get name of close command."""
+        _LOGGER.info("def close_cmd called=%s", self._close_cmd)
+        return self._close_cmd
+
+    @property
+    def stop_cmd(self):
+        """Get name of stop command."""
+        _LOGGER.info("def close_cmd called=%s", self._stop_cmd)
+        return self._stop_cmd
+
+    @property
+    def last_movement_dps(self):
+        """Get last movement key"""
+        #_LOGGER.info("def last_movement_dps called=%s", self._last_movement_dps)
+        return self._last_movement_dps
+
+
+    @property
+    def last_movement(self):
+        #_LOGGER.info("def last_movement called=%s", self._last_movement)
+        """Return the last movement of the device (should be "Opening" or "Closing")"""
+        return self._last_movement
+
+    def last_position_set(self):
+        #_LOGGER.info("def last_position_set called=%s", self._last_position_set)
+        """Return the last position set of the device"""
+        return self._last_position_set
+        
+    @property
+    def last_command(self):
+        #_LOGGER.info("def last_command called=%s", self._last_command)
+        """Return the last command of the device"""
+        return self._last_command
+
+    @property
+    def is_open(self):
+        """Check if the cover is partially or fully open."""
+        #_LOGGER.info("def is_open called, self._current_cover_position = %s", self._current_cover_position)
+        if self._current_cover_position != 100:
+            #_LOGGER.info("is_open returning true)")
+            return True
+        #_LOGGER.info("is_open returning false")
+        return False
+
+    @property
+    def is_closed(self):
+        """Check if the cover is fully closed."""
+        #_LOGGER.info("def is_closed called")
+        if self._current_cover_position == 100:
+            #_LOGGER.info("is_closed returning true)")
+            return True
+        #_LOGGER.info("is_closed returning false")
+        return False
+
+    def update(self):
+        """Update cover attributes and store in cache."""
+        _LOGGER.info("update(self) called")
+        try:
+            _LOGGER.info("about to call status=self._device.status() from def update(self)")
+            status = self._device.status()
+            _LOGGER.info("def update set status =%s", status)
+            self._cached_status = status
+            _LOGGER.info("self._cached_status set =%s", self._cached_status)
+            
+            _LOGGER.info("about to call self._update_last_command()")
+            self._update_last_command()
```

Comment:
I believe that you can just inline the variable assignments of al these calls here. The names of the variables make it quite clear what's going on (at least the methods doesn't add much). If you really need debugging here, add one line that prints every value (compactly) with `debug` level. It gets very chatty otherwise.